featurebyte/service/feature_materialize_scheduler.py

```diff
@@ -62,15 +68,21 @@ async def start_job_if_not_exist(
             try:
                 if offline_store_feature_table.feature_job_setting is None:
                     return
-                await self.task_manager.schedule_interval_task(
-                    name=self._get_job_id(offline_store_feature_table.id),
-                    payload=payload,
-                    interval=Interval(
-                        every=offline_store_feature_table.feature_job_setting.period_seconds,
-                        period="seconds",
-                    ),
-                    time_modulo_frequency_second=offline_store_feature_table.feature_job_setting.offset_seconds,
-                )
+                if isinstance(offline_store_feature_table.feature_job_setting, FeatureJobSetting):
+                    await self.task_manager.schedule_interval_task(
+                        name=self._get_job_id(offline_store_feature_table.id),
+                        payload=payload,
+                        interval=Interval(
+                            every=offline_store_feature_table.feature_job_setting.period_seconds,
+                            period="seconds",
+                        ),
+                        time_modulo_frequency_second=offline_store_feature_table.feature_job_setting.offset_seconds,
+                    )
+                else:
+                    # DEV-3924: this method should be implemented for cron feature job setting
+                    raise NotImplementedError(
+                        f"Feature job setting type {type(offline_store_feature_table.feature_job_setting)} is not supported"
+                    )
```

Comment:
Would it make sense to raise a temporary but more specific exception such as `CronNotImplementedError` so that it's eaiser to look up their occurences?